src/System Application/App/File System/src/Lookup/FileAccountBrowser.Page.al

```diff
@@ -0,0 +1,304 @@
+                    begin
+                        case true of
+                            (Rec.Name = '..'):
+                                BrowseFolder(Rec."Parent Directory");
+                            (Rec.Type = Rec.Type::Directory):
+                                BrowseFolder(Rec);
+                            (not IsInLookupMode):
+                                DownloadFile(Rec);
+                        end;
+                    end;
+                }
+                field("Type"; Rec."Type")
+                {
+                    ToolTip = 'Specifies the value of the Type field.';
+                }
+            }
+
+            group(SaveFileNameGroup)
+            {
+                Caption = '', Locked = true;
+                ShowCaption = false;
+                Visible = ShowFileName;
+
+                field(SaveFileNameField; SaveFileName)
+                {
+                    Caption = 'Filename';
+                }
+            }
+        }
+    }
+
+    actions
+    {
+        area(Promoted)
+        {
+            actionref(UploadRef; Upload) { }
+            actionref(CreateDirectoryRef; "Create Directory") { }
+            actionref(DeleteRef; Delete) { }
+        }
+        area(Processing)
+        {
+            action(Upload)
+            {
+                Caption = 'Upload';
+                Image = Import;
+                Ellipsis = true;
+                Visible = not IsInLookupMode;
+                Enabled = not IsInLookupMode;
+
+                trigger OnAction()
+                begin
+                    UploadFile();
+                    BrowseFolder(CurrPath);
+                end;
+            }
+            action("Create Directory")
+            {
+                Caption = 'Create Directory';
+                Image = Bin;
+                Ellipsis = true;
+                Visible = not IsInLookupMode;
+                Enabled = not IsInLookupMode;
+
+                trigger OnAction()
+                begin
+                    CreateDirectory();
+                    BrowseFolder(CurrPath);
+                end;
+            }
+            action(Delete)
+            {
+                Caption = 'Delete';
+                Image = Delete;
+                Ellipsis = true;
+                Visible = not IsInLookupMode;
+                Enabled = not IsInLookupMode;
+
+                trigger OnAction()
+                begin
+                    DeleteFileOrDirectory();
+                    BrowseFolder(CurrPath);
+                end;
+            }
+        }
+    }
+
+    var
+        FileSystem: Codeunit "File System";
+        CurrPath, CurrFileFilter, SaveFileName, CurrPageCaption : Text;
+        ParentFolderExists, DoNotLoadFields, IsInLookupMode, ShowFileName : Boolean;
+
+    trigger OnOpenPage()
+    begin
+        if CurrPageCaption <> '' then
+            CurrPage.Caption(CurrPageCaption);
+    end;
+
+    internal procedure SetFileAcconut(FileAccount: Record "File Account")
```

Comment:
Pages should only contain the minimum amount of UI related code as is required. There should not be any logic here. Any logic should be placed in codeunits.